tests/test_process.py

```diff
@@ -90,6 +91,248 @@ def test_upload_dataversion_metadata_without_team_images_id(self, syn: Any):
         )
 
 
+class TestApplyCustomTransformations:
+    """Test the apply_custom_transformations function."""
+
+    @pytest.fixture(scope="function", autouse=True)
+    def standard_transform_function(
+        self,
+    ) -> Callable[[pd.DataFrame, str, Dict[str, pd.DataFrame]], pd.DataFrame]:
+        """mock simple transform function that uses standard parameters such as dataframe, dataset_name, datasets"""
+
+        def _standard_transform_function(
+            df: pd.DataFrame, dataset_name: str, datasets: Dict[str, pd.DataFrame]
+        ) -> pd.DataFrame:
+            """mock simple transform function"""
+            if dataset_name in datasets:
+                aggregated_value = datasets[dataset_name]["value_column"].sum()
+                df["aggregated_value"] = aggregated_value
+            return df.assign(new_key=dataset_name + "_new_key")
+
+        return _standard_transform_function
+
+    @pytest.fixture(scope="function", autouse=True)
+    def special_transform_function(
+        self,
+    ) -> Callable[[pd.DataFrame, int, Dict[str, pd.DataFrame]], pd.DataFrame]:
+        """mock transform function that uses additional parameters"""
+
+        def _mock_transform_with_args(
+            df: pd.DataFrame, test_threshold: int
+        ) -> pd.DataFrame:
+            """mock transform function that takes test_threshold as an argument"""
+            return df.assign(new_key=test_threshold)
+
+        return _mock_transform_with_args
+
+    @pytest.mark.parametrize(
+        "example_config_with_custom_transform,expectation",
+        # Fails because mock_transform_function_false doesn't exist
+        [
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": {
+                            "mock_transform_function_false": {
+                                "test_threshold": 1,
+                                "test_p_value": 2,
+                            }
+                        },
+                    }
+                },
+                pytest.raises(AttributeError),
```

Comment:
Let's make sure we are using `match=` for any custom error messages too where applicable

Reply:
Okay sounds good.